### PR TITLE
[efr32] Update for compatibility with GSDK 3.0

### PR DESCRIPTION
**Pull request by lmnotran**
This PR includes a relatively minimal set of changes that are required to allow the efr32 platforms to function with the newly released Silabs Gecko SDK v3.0. 

# Change Log

- Pull in definitions for `halInitChipSpecific()` and all child functions from GSDK v2.7.
  - These will be replaced with a more elegant solution in a future PR.
- Switch to the mbedtls that is packaged with GSDK v3.0.
- Allow for mixed casing when specifying `BOARD` environment variable

### CODE REVIEW

File: examples/platforms/efr32mg13/system.c [hunk 1]
```diff
@@ -55,7 +58,6 @@
 #include "sl_mpu.h"
 #include "sl_sleeptimer.h"
 
-#include "openthread-core-efr32-config.h"
```
Comment (lmnotran) [inline examples/platforms/efr32mg13/system.c:58]: This is included in `#include <openthread-core-config.h>	`

**Review by romacdon — CHANGES_REQUESTED** [review]

**Review by romacdon — APPROVED** [review]

**Comment by romacdon** [comment]
@jwhui I approved this PR but I would like @schinchani to review as well too but I can seem to assign him as a reviewer.

**Comment by jwhui** [comment]
> @jwhui I approved this PR but I would like @schinchani to review as well too but I can seem to assign him as a reviewer.

@romacdon , I updated @schinchani 's config. Can you try again?

**Comment by romacdon** [comment]
> > @jwhui I approved this PR but I would like @schinchani to review as well too but I can seem to assign him as a reviewer.
> 
> @romacdon , I updated @schinchani 's config. Can you try again?

Thanks for the unbelievably speedy reply!  I was able to add him as a reviewer.  Can ask what config you needed to change in case I need to something similar for other reviewers in the future?

**Comment by jwhui** [comment]
> Thanks for the unbelievably speedy reply! I was able to add him as a reviewer. Can ask what config you needed to change in case I need to something similar for other reviewers in the future?

@romacdon , I needed to add him to the `openthread/silicon-labs` group on GitHub.

**Review by romacdon — APPROVED** [review]

**Review by schinchani — APPROVED** [review]
Looks good to me @lmnotran

**Pull request merged**